extensions/ql-vscode/src/databases-ui.ts

```diff
@@ -670,7 +718,11 @@ export class DatabaseUI extends DisposableObject {
 }
 
 // TODO: Get the list of supported languages from a list that will be auto-updated.
-const dbRegeEx = /^db-(javascript|go|cpp|java|python|csharp)$/;
+const dbRegeEx = /^db-(javascript|go|cpp|java|python|csharp|ruby)$/;
 function isLikelyDbFolder(dbPath: string) {
   return path.basename(dbPath).match(dbRegeEx);
 }
+
+async function isDatabaseDirectory(dir: string) {
+  return (await fs.readdir(dir)).some(isLikelyDbFolder);
+}
```

Comment:
Not very fond of having to maintain a list of language slugs.
Elsewhere, I expect we are checking for the existence of `codeql-database.yml` (or `.dbinfo` as a fallback) to determine whether a directory is a CodeQL database. Can we continue to use that here instead of relying on the dataset folder name?

Reply:
I agree about the hard-coded languages, but I'm not quite sure what to do about them. Maybe there is some kind of way we can discover this by introspecting which standard libraries are installed, but that seems complex for now. 

I'll be more precise and check for one of those two files.